[SDK-3289] Update miscellaneous references and code snippets for Swift SDK (#5235)
braze-inc/braze-docs: _docs/_developer_guide/platform_wide/sdk_authentication.md

This feature is available as of the following [SDK versions]({{ site.baseurl }}/user_guide/engagement_tools/campaigns/ideas_and_strategies/new_features/#filtering-by-most-recent-app-versions):

{% sdk_min_versions web:3.3.0 ios:5.0.0 android:14.0.0 %}

{% alert note %}

> For iOS integrations, this page details the steps for the Braze Swift SDK. For sample usage in the legacy AppboyKit iOS SDK, reference [this file](https://github.com/Appboy/appboy-ios-sdk/blob/master/Example/Stopwatch/Sources/AppDelegate.m).

Reviewer: ```suggestion
For iOS integrations, this page details the steps for the Braze Swift SDK. For sample usage in the legacy AppboyKit iOS SDK, reference [this file](https://github.com/Appboy/appboy-ios-sdk/blob/master/Example/Stopwatch/Sources/AppDelegate.m) and [this file](https://github.com/Appboy/appboy-ios-sdk/blob/master/Example/Stopwatch/Sources/Utils/SdkAuthDelegate.m).
```

This feature is available as of the following [SDK versions]({{ site.baseurl }}/user_guide/engagement_tools/campaigns/ideas_and_strategies/new_features/#filtering-by-most-recent-app-versions):

> {% sdk_min_versions web:3.3.0 ios:5.0.1 android:14.0.0 %}

Author: This linking seems to be broken for some reason. I'm not sure what's going on here.

Alternatively, you can add `<bool name="com_braze_sdk_authentication_enabled">true</bool>` to your braze.xml.
{% endtab %}
{% tab Objective-C %}
To enable SDK Authentication, set the `configuration.api.sdkAuthentication` property of your `BRZConfiguration` object to `YES` before initializing the Braze instance:

```objc
BRZConfiguration *configuration =
    [[BRZConfiguration alloc] initWithApiKey:@"{BRAZE_API_KEY}"
                                    endpoint:@"{BRAZE_ENDPOINT}"];
configuration.api.sdkAuthentication = YES;
Braze *braze = [[Braze alloc] initWithConfiguration:configuration];
AppDelegate.braze = braze;
```
{% endtab %}
{% tab Swift %}
To enable SDK Authentication, set the `configuration.api.sdkAuthentication` property of your `Braze.Configuration` object to `true` when initializing the SDK:

```swift
let configuration = Braze.Configuration(apiKey: "{YOUR-BRAZE-API-KEY}",
                                        endpoint: "{YOUR-BRAZE-ENDPOINT}")
configuration.api.sdkAuthentication = true
let braze = Braze(configuration: configuration)
AppDelegate.braze = braze

> ```

Reviewer: Bump for [this comment](https://github.com/braze-inc/braze-docs/pull/5235#discussion_r1164524410) about including a way to access the example code for AppboyKit if they are using the legacy SDK

Author: I included it [here](https://github.com/braze-inc/braze-docs/pull/5235/files#diff-dc9bf7157a256faa0d5917fbe50764070967ea3b951c1647eb808c2b90411e12R80-R84) at the top-level instead of at each code snippet - let me know what you think!

```
{% endtab %}
{% tab Objective-C %}

{% alert important %}

> Starting in version `5.14.0` of the Braze Swift SDK, the SDK authentication delegate method has been split from the `BrazeDelegate` into a separate `BrazeSDKAuthDelegate`.

Author: This change is pending release, so we'll need to coincide this merge with the release of the new version.

Reviewer 2: @jerielng Do you know the time frame of the release?

Author: It'll be within the next week. We have most of the changes ready, just doing some final merges and reviews.

```
{% endtab %}
{% tab Objective-C %}

{% alert important %}
Starting in version `5.14.0` of the Braze Swift SDK, the SDK authentication delegate method has been split from the `BrazeDelegate` into a separate `BrazeSDKAuthDelegate`.
{% endalert %}

```objc
Braze *braze = [[Braze alloc] initWithConfiguration:configuration];
braze.sdkAuthDelegate = delegate;
AppDelegate.braze = braze;

// Method to implement in delegate
- (void)braze:(Braze *)braze sdkAuthenticationFailedWithError:(BRZSDKAuthenticationError *)error {
  // TODO: Optionally log to your error-reporting service
  // TODO: Check if the `user_id` within the `error` matches the currently logged-in user
  NSLog(@"Invalid SDK Authentication signature.");
  NSString *newSignature = getNewSignatureSomehow(error);
  [AppDelegate.braze setSDKAuthenticationSignature:newSignature];
}
```
{% endtab %}
{% tab Swift %}

{% alert important %}

> Starting in version `5.14.0` of the Braze Swift SDK, the SDK authentication delegate method has been split from the `BrazeDelegate` into a separate `BrazeSDKAuthDelegate`.

Reviewer 3: (do we have any additional sections or docs to link to for this?)

Author: The integration instructions in this doc describe the usage of the new `BrazeSDKAuthDelegate` with the assumption that these Swift docs will be released post-`5.14.0`, so anyone referencing this should be referencing the new protocol. This note is to indicate the split in case anyone is on an older version and might get confused. I could modify this to say:
```suggestion
Starting in version `5.14.0` of the Braze Swift SDK, the SDK authentication delegate method has been split from the `BrazeDelegate` into a separate `BrazeSDKAuthDelegate`. If you are on a version prior to this, the SDK authentication delegate method should be referenced from your implementation of `BrazeDelegate`.
```

Any thoughts?

Reviewer 3: 👍 yeah, that sounds good; should `from` there be `for`?

```suggestion
Starting in version `5.14.0` of the Braze Swift SDK, the SDK authentication delegate method has been split from the `BrazeDelegate` into a separate `BrazeSDKAuthDelegate`. If you are on a version prior to this, you should reference the SDK authentication delegate method for your implementation of `BrazeDelegate`.
```

Author: Ah maybe the wording was a bit confusing. The previous version of the SDK auth delegate method lives inside the `BrazeDelegate`, so what I was trying to convey was you should call the method from whatever implementation they have of `BrazeDelegate` instead of `BrazeSDKAuthDelegate` - let me know if there's a better way to phrase this 😅

Reviewer 3: ah ah ah ok yep. "wherever you conform t `BrazeDelegate`" maybe?

Author: ```suggestion
Starting in version `5.14.0` of the Braze Swift SDK, the SDK authentication delegate method has been separated from the `BrazeDelegate` into the `BrazeSDKAuthDelegate` protocol. If you are on a version prior to this, you should implement the SDK authentication delegate method wherever you conform to `BrazeDelegate`.
```
